yt_dlp/extractor/downdogapp.py

```diff
@@ -0,0 +1,219 @@
+import time
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    urlencode_postdata,
+)
+
+
+class DownDogAppGenericIE(InfoExtractor):
+    # Generic extractor, only used to instantiate subclasses
+
+    def __init_subclass__(cls, /, subdomain, **kwargs):
+        super().__init_subclass__(**kwargs)
+        cls._SUBDOMAIN = subdomain
+        cls._COOKIE_DOMAIN = f'{subdomain}.downdogapp.com'
+        cls._VALID_URL = rf'https?://{subdomain}\.downdogapp\.com/play#(?P<id>[^\?]*)'
+        cls._LOGIN_URL = f'https://{subdomain}.downdogapp.com/json/login'
+        cls._MANIFEST_URL = rf'https://{subdomain}.downdogapp.com/manifest'
+        cls._NETRC_MACHINE = f'downdogapp{subdomain}'
+
```

Comment:
Do you really need separate accounts for each subdomain? If not, this should be a single extractor.

(And even if you need separate accounts for each subdomain, this can still be a single extractor; see `yt_dlp.extractor.getcourseru` for an example)